### DISCUSSION TITLE
Code for surfactant flooding

### DISCUSSION THREAD

**Discussion by Kevin9384** [PMEAL/OpenPNM#1897] [Q&A]
Hello everyone, I'm currently studying petroleum engineering and having a project. I'm new to OpenPNM and I would like to know how to write the code for testing the effect of concentration of surfactant to surfactant flooding.

**Comment by jgostick** [comment]
What is the 'effect' you're looking for?  I'm guessing that you'll be doing some 2-phase flow stuff, and want to see how the fluid configuration evolves after injecting some surfactant into to water phase?  Before I ponder this too much, I'll wait for you to provide more detail.

**Reply by Kevin9384** [reply]
Effect on the adsorption rate. Can I visualize the advection, diffusion and dispersion of the surfactant on formation using Open PNM?

**Reply by jgostick** [reply]
yes, if you just want to see where the surfactant will go with time you can use our AdvectionDiffusion algorithm.  [Here is an example](https://github.com/PMEAL/OpenPNM/blob/dev/examples/notebooks/algorithms/transient/transient_advection_diffusion.ipynb)
